src/opnsense/scripts/filter/lib/alias/__init__.py

```diff
@@ -164,36 +164,51 @@ def resolve(self, force=False):
             :return: string
         """
         if not self._resolve_content:
+
             if self.expired() or self.changed() or force:
-                if os.path.isfile(self._filename_alias_content):
+                alias_file_exists = os.path.isfile(self._filename_alias_content)
+                if alias_file_exists:
                     try:
                         undo_content = open(self._filename_alias_content, 'r').read()
                     except UnicodeDecodeError:
                         undo_content = ""
                 else:
                     undo_content = ""
+                address_parser = self.get_parser()
+                if os.path.isfile(self._filename_alias_hash):
```

Comment:
I think I would move this into the `try..except` block on top, if content isn't usable, it's safe to assume the hash has no value too. The files we write are also highly unlikely to fail on a unicode decode error.